pkg/client/client.go

```diff
@@ -0,0 +1,410 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package client
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"sync"
+	"time"
+
+	"google.golang.org/grpc"
+
+	"github.com/elastic/elastic-agent-client/v7/pkg/proto"
+	"github.com/elastic/elastic-agent-client/v7/pkg/utils"
+)
+
+// CheckinMinimumTimeout is the amount of time the client must send a new checkin even if the status has not changed.
+const CheckinMinimumTimeout = time.Second * 30
```

Comment:
I wonder if we should use a lower value here to be able to report faster if a process is not active anymore?

Reply:
Note: I think this value should be configurable at runtime, for lets say that you want more real time response from the client.

Reply:
I just reread the above, I don't think my comment make sense in that context.

Reply:
This is only used when the current status of the application has not changed so if the application sets the status to `HEALTHY` that will be sent immediately at that point (minus some debounce of 500ms). Then it will keep sending `HEALTHY` every 30 seconds.

Being that Elastic Agent will have an expectation of check-in time, I don't think we need to have this configurable.

Reply:
@blakerouse sound good to me.